builder.ts

```diff
@@ -0,0 +1,223 @@
+import { input, select } from "@inquirer/prompts";
+import { Contract, FunctionFragment, getDefaultProvider, isAddress, isHexString, parseUnits, toBigInt } from "ethers";
+import { TransactionType, encodeMulti, MetaTransaction, OperationType, encodeSingle } from "ethers-multisend";
+import { spawnSync } from "child_process";
+import { writeFileSync } from "fs";
+import * as path from "path";
+import { config } from "dotenv";
+
+config();
+
+const operationTypes = {
+  Call: OperationType.Call,
+  DelegateCall: OperationType.DelegateCall,
+};
+
+const trial =
+  (transformer: (value: string) => string | boolean | Promise<string | boolean>, error: string) =>
+  async (value: string) => {
+    try {
+      return (await transformer(value)) || error;
+    } catch (err: any) {
+      return `${error} "${value}": ${err.message}`;
+    }
+  };
+
+const address = (config: { message: string; default?: string }) =>
+  input({
+    ...config,
+    validate: trial((value) => isAddress(value), "Invalid address"),
+  });
+
+const number = (config: { message: string; min: number; default?: string }) =>
```

Comment:
Not a fan of functions that have the name of a type
```suggestion
const inputNumber = (config: { message: string; min: number; default?: string }) =>
```